gabbi/driver.py

```diff
@@ -132,6 +133,12 @@ def build_tests(path, loader, host=None, port=8001, intercept=None,
             else:
                 suite_dict['defaults'] = {'verbose': verbose}
 
+        if any((cert_validate == opt for opt in [True, False])):
```

Comment:
Since `cert_validate` is a boolean and we want any arg type issues to bubble up quickly when someone is building their test system, and since the default is already set in the signature of the containing method (`build_tests`) I think you can model the code here on what's at line 123 and used by `require_ssl`:

```python
if not cert_validate:
    if 'defaults' in suite_dict:
       suite_dict['defaults']['cert_validate'] = cert_validate
    else:
        suite_dict['defaults'] = {'cert_validate': cert_validate}
```

or something like that. Similar comments apply in other places that you've used the same idiom.